libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -639,9 +661,30 @@ private void removeFpsListenersFromCamera() {
     mapCamera.removeOnTrackingModeChangedListener(mapFpsDelegate);
   }
 
-  private void updateMapWaynameWithLocation(Location location) {
+  private void updateMapWayNameWithLocation(Location location) {
     LatLng latLng = new LatLng(location);
     PointF mapPoint = mapboxMap.getProjection().toScreenLocation(latLng);
     mapWayName.updateWayNameWithPoint(mapPoint);
   }
+
+  private void restoreMapWith(NavigationMapSettings settings) {
+    updateCameraTrackingMode(settings.retrieveCameraTrackingMode());
+    if (settings.shouldUseDefaultPadding()) {
```

Comment:
What about having separate `settings` objects that reside where they are used? I.e., fps settings would reside in the `mapFpsDelegate`, and wayname settings could reside in `mapWayName`, etc.